src/octoprint/util/json/__init__.py

```diff
@@ -1,46 +1,13 @@
 __license__ = "GNU Affero General Public License http://www.gnu.org/licenses/agpl.html"
-__copyright__ = "Copyright (C) 2018 The OctoPrint Project - Released under terms of the AGPLv3 License"
+__copyright__ = "Copyright (C) 2022 The OctoPrint Project - Released under terms of the AGPLv3 License"
 
+from octoprint.util import deprecated
 
-import collections
-import json
+from . import serializing  # noqa: F401
+from .encoding import JsonEncoding, dumps, loads  # noqa: F401
 
-from frozendict import frozendict
-
-from octoprint.util import to_unicode
-
-
-def dump(obj):
-    return json.dumps(
-        obj,
-        separators=(",", ":"),
-        default=JsonEncoding.encode,
-        allow_nan=False,
-    )
-
-
-class JsonEncoding:
-
-    encoders = collections.OrderedDict()
-
-    @classmethod
-    def add_encoder(cls, type, encoder):
-        cls.encoders[type] = encoder
-
-    @classmethod
-    def remove_encoder(cls, type):
-        try:
-            del cls.encoders[type]
-        except KeyError:
-            pass
-
-    @classmethod
-    def encode(cls, obj):
-        for type, encoder in cls.encoders.items():
-            if isinstance(obj, type):
-                return encoder(obj)
-        raise TypeError
-
-
-JsonEncoding.add_encoder(frozendict, lambda obj: dict(obj))
-JsonEncoding.add_encoder(bytes, lambda obj: to_unicode(obj))
+dump = deprecated(
```

Comment:
is this module part of the public octoprint API?

Reply:
Not per se documented, but as with everything under util, `dump` might be in use by third party plugins and I prefer to at least give them a warning before just throwing stuff out. Was still even used in core OctoPrint itself btw, also fixed that.